src/Go.CodeAnalysis/Text/SnapshotBase.cs

```diff
@@ -1,16 +1,30 @@
 ﻿namespace Go.CodeAnalysis.Text
 {
+    /// <summary>
+    /// Abstract representation of a text segment.
```

Comment:
Perhaps worth mentioning that this is an immutable snapshot of a document at a point in time.